internal/pkg/cli/deploy/backend.go

```diff
@@ -20,8 +20,11 @@ import (
 
 type backendSvcDeployer struct {
 	*svcDeployer
-	backendMft         *manifest.BackendService
+	backendMft *manifest.BackendService
+
+	// Overriden in tests.
 	aliasCertValidator aliasCertValidator
+	newStack           func() cloudformation.StackConfiguration
```

Comment:
What do you think of doing something similar with `cli/deploy/env.go`?
```go
// In cli/deploy/backend.go
type  backendSvcDeployer struct {
    newStack func() cloudformation.StackConfiguration
}

func NewBackendDeployer(in *WorkloadDeployerInput) (*backendSvcDeployer, error) {
    return &backendSvcDeployer{
		newStack: func() cloudformation.StackConfiguration {
                       return stack.NewBackendService(stack.BackendServiceConfig{
			App:                d.app,
			EnvManifest:        d.envConfig,
			Manifest:           d.backendMft,
			RawManifest:        d.rawMft,
			ArtifactBucketName: d.resources.S3Bucket,
			RuntimeConfig:      *rc,
			Addons:             d.addons,
		     })
                 }
	}, nil 
}

// In cli/deploy/backend_test.go
func TestBackendServiceDeployer_stackConfiguration(t *testing.T) {
   ....
   stackSerializer: cfnmocks.NewMockStackConfiguration(ctrl),
   deployer := &backendSvcDeployer{...}
   deployer.newStack = func () cloudformation.StackConfiguration{ return stackSerializer }
}

```

Reply:
I wish :cool-crying: if you look at the lbws instead of backend you can see that it's not possible :'(

I think the real solution is to use dependency injection properly in both `cli/deploy` and `stack`. Then I could have just passed down some stubs and we wouldn't need to write a `newStack` field in the first place